spec/yaml/swiftnav/sbp/piksi.yaml

```diff
@@ -521,6 +523,7 @@ definitions:
 
  - MSG_CELL_MODEM_STATUS:
     id: 0x00BE
+    message_display: "{{@ signal_strength @}} dBm | {{@ signal_error_rate @}} %"
```

Comment:
I went with a funky regex group capturing

`abc{{X}}` represents `format!("abc{}", X)`
`abc{{@ X @}}` represents `format!("abc{}", self.X)`

Basically `{{@ ... @}}` prepends `self.`
This allows some things that only requires fields to be simplified but might be bit hacky.

Missing functionality to do something like this `format!("{:.2}", self.X)` 
Maybe TODO: add more parsing for some syntaxes like `{{@ ... @}:.3}` where we also retain `{{}:.}` some element after char `:`.

We can parse methods if there are by just calling them like `{{@ some_method() @}}` etc... but maybe really hacky to be inside yaml

Reply:
The first one could be used something like 
```
{{@ posX @ / 10}}
format!({}, self.posX / 10)
```

There are cases where we do [operations](https://github.com/swift-nav/gt/blob/e9e97373532aa7f39c94cdd7a2abc0c1ac72b12c/sw/mod/proto/sbp/gt_proto_sbp_csv.c#L975) operations on these fields - which might turn out really hacky inside the yaml when it gets really complicated as something like

```
{{if @ bool @ {5} else { 10 }}}
format!("{}", if self.bool { 5 } else { 10 })
```

Maybe there is a better way since these format methods would turn out really funky codegen wise, but avoided manual implementations of `message_display` by these scripting formats

Reply:
But yea `@field_name` style might be nice but would be weird to decide where to terminate i.e.
```rust
{{@posX + @posY }} // requiring end char to be ' '
{{@posX+@posY}} // no ' ' termination character
```
*Having `@field_name` might be simpler but was unsure if '@' came up often so might swap to this style instead

Reply:
Ah okay I see.  These YAMLs probably want to be language independent right, we would not be able to use `message_display` in the python library if it is a rust expression.  I think until there are usecases for this it might be simpler to only allow simple string interpolation for now.

Reply:
I think we can use {fmt} style interpolation in most languages, but it'd take some work, and it'd need to be done in a language neutral way, so we couldn't have embedded Rust/whatever syntax:
- Java: https://fmt.dev/5.2.1/index.html
- C++: https://github.com/fmtlib/fmt
- Rust (obviously): https://doc.rust-lang.org/std/fmt/
- https://www.w3schools.com/python/ref_string_format.asp
- JavaScript? https://www.npmjs.com/package/format-unicorn

This feature would not work for languages that don't implement some variant of the {fmt} string interpolation.